master/buildbot/www/authz/roles.py

```diff
@@ -70,3 +70,15 @@ def getRolesFromUser(self, userDetails, owner):
             if userDetails['email'] == owner and owner is not None:
                 return [self.role]
         return []
+
+
+class RolesFromUsername(RolesFromBase):
+    def __init__(self, roles, usernames):
+        self.roles = roles
+        self.usernames = usernames
+
+    def getRolesFromUser(self, userDetails):
+        if 'username' in userDetails:
```

Comment:
I would prefer that you ensure None not in usernames in the constructor (with a config,error)